src/heretic/r_data.c

```diff
@@ -554,6 +619,31 @@ void R_InitColormaps(void)
     }
 }
 
+#ifdef CRISPY_TRUECOLOR
+// [crispy] Changes palette to E2PAL. Used exclusively in true color rendering
+// for proper drawing of E2END pic in F_DrawUnderwater. Changing palette back to
+// original PLAYPAL for restoring proper colors will be done in R_InitColormaps.
+void R_SetUnderwaterPalette(void)
```

Comment:
This function would be more universally usable (for truecolor Hexen, maybe) if you passed the palette lump name as an argument.

Reply:
Fixed in https://github.com/fabiangreffrath/crispy-doom/pull/1111/commits/00c44aa88c3d0bcedb399ae710433e3403165d4e. It's still local as discussed, but passing parametrized lump name was a very good idea:
* This makes code much neater and w/o extra macroses.
* `E2PAL` seems to be HeHacked`able name, so we have to care about it's possible name replaced, even though such replacement probably will never happen.